### PR TITLE
Gather iOS device logs after running tests

### PR DESCRIPTION
**Pull request by PureWeen**

### CODE REVIEW

File: eng/pipelines/common/ui-tests-compatibility-steps.yml [hunk 1]
```diff
@@ -44,7 +44,16 @@ steps:
   - pwsh: ./build.ps1 -Script eng/devices/${{ parameters.platform }}.cake --target=cg-uitest --project="${{ parameters.path }}" --appproject="${{ parameters.app }}" --device="${{ parameters.device }}" --apiversion="${{ parameters.version }}" --configuration="${{ parameters.configuration }}" --results="$(TestResultsDirectory)" --binlog="$(LogDirectory)" ${{ parameters.cakeArgs }} --verbosity=diagnostic --where="${{ parameters.where }}"
     displayName: $(Agent.JobName)
     # retryCountOnTaskFailure: 2
-  
+
+  - bash: |
+      rm -r $HOME/Library/Logs/CoreSimulator/*
+      rm -r $HOME/Library/Logs/DiagnosticReports/*
+      zip -9r "$(LogDirectory)/artifacts/CoreSimulatorLog.zip" "$HOME/Library/Logs/CoreSimulator/"
+      zip -9r "$(LogDirectory)/DiagnosticReports.zip" "$HOME/Library/Logs/DiagnosticReports/"
```
Comment (PureWeen) [inline eng/pipelines/common/ui-tests-compatibility-steps.yml:52]: do we need to suffix a random string so we don't overwrite things on multiple runs?

File: eng/pipelines/common/device-tests-steps.yml [hunk 2]
```diff
@@ -86,6 +86,17 @@ steps:
     condition: and(succeeded(), eq(variables['Platform.Name'], 'Mac'))
     retryCountOnTaskFailure: 1
 
+
+  - ${{ if eq(parameters.platform, 'ios')}}:
+    - bash: |
+        rm -r $HOME/Library/Logs/CoreSimulator/*
+        rm -r $HOME/Library/Logs/DiagnosticReports/*
+        zip -9r "$(LogDirectory)/CoreSimulatorLog.zip" "$HOME/Library/Logs/CoreSimulator/"
+        zip -9r "$(LogDirectory)/DiagnosticReports.zip" "$HOME/Library/Logs/DiagnosticReports/"
```
Comment (rolfbjarne) [inline eng/pipelines/common/device-tests-steps.yml:95]: It's early in the morning for me here, but aren't you first deleting the directories, and then zipping them up? How is that going to gather anything at all?

Comment (PureWeen) [reply eng/pipelines/common/device-tests-steps.yml:95]: aw crap
I really didn't think through this at all did I :-/

File: eng/pipelines/common/device-tests-steps.yml [hunk 3]
```diff
@@ -71,6 +71,14 @@ steps:
       Write-Host "##vso[task.setvariable variable=Platform.Name]${platformName}"
     displayName: 'Set Platform.Name'
 
+  - ${{ if eq(parameters.platform, 'ios')}}:
+    - bash: |
+        rm -r $HOME/Library/Logs/CoreSimulator/*
+        rm -r $HOME/Library/Logs/DiagnosticReports/*
+      displayName: Zip Simulator Logs
```
Comment (mattleibow) [inline eng/pipelines/common/device-tests-steps.yml:78]:
```suggestion
      displayName: Delete Old Simulator Logs
```

File: eng/pipelines/common/ui-tests-steps.yml [hunk 4]
```diff
@@ -99,10 +99,26 @@ steps:
   - pwsh: ./build.ps1 --target=dotnet-samples --configuration="${{ parameters.configuration }}" --${{ parameters.platform }} --verbosity=diagnostic --usenuget=false
     displayName: 'Build the samples'
 
+  - ${{ if eq(parameters.platform, 'ios')}}:
+    - bash: |
+        rm -r $HOME/Library/Logs/CoreSimulator/*
+        rm -r $HOME/Library/Logs/DiagnosticReports/*
+        displayName: Zip Simulator Logs
```
Comment (mattleibow) [inline eng/pipelines/common/ui-tests-steps.yml:106]:
```suggestion
        displayName: Delete Old Simulator Logs
```

File: eng/pipelines/common/ui-tests-compatibility-steps.yml (hunk 1 above)
Comment (mattleibow) [reply eng/pipelines/common/ui-tests-compatibility-steps.yml:57]: we probably do. Maybe the date time stuff? yyyyMMddHHmmss

Comment (mattleibow) [reply eng/pipelines/common/ui-tests-compatibility-steps.yml]: @PureWeen  we need the hours and mins. A retry will typically happen in the same hour.

Comment (PureWeen) [reply eng/pipelines/common/ui-tests-compatibility-steps.yml]: yea, I think you caught me between pushes :-) 
Should be good now

**Review by mattleibow — APPROVED** [review]

**Pull request merged**